src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -42,18 +35,24 @@ ImmutabilityContext context
 				// definition. Although we care about them otherwise, TODO explain
 				.Where( m => !m.IsStatic );
 
-			// Check that the base class is immutable
-			var baseClassOk = m_context.IsImmutable(
-				type.BaseType,
-				ImmutableTypeKind.Instance,
-				decl.BaseList?.Types.First()?.GetLocation(),
-				out var diag
-			);
+			// Check that the base class is immutable for classes
+			// (every user-defined class ultimately extends System.Object; stop there.)
+			if ( type.TypeKind == TypeKind.Class && type.BaseType?.SpecialType != SpecialType.System_Object ) {
+				// TODO: put this on the BaseTypeSyntax instead

```

Comment:
I'm pretty sure what I've got for this works but wanted to give you a PR that definitely wouldn't crash first (no tests).
I'll open another PR with the impl.